src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/DelegateHelpers.cs

```diff
@@ -4,14 +4,15 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
 using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
 using System.Text;
 using System.Threading;
 
 namespace System.Dynamic.Utils
 {
     internal static class DelegateHelpers
     {
-        // This can be flipped to true using feature switches at publishing time
+        // This can be flipped to false using feature switches at publishing time
         internal static bool CanEmitObjectArrayDelegate => true;
```

Comment:
Would it make sense to change this into
```csharp
internal static bool CanEmitObjectArrayDelegate => RuntimeFeature.CanEmitObjectArrayDelegate;
```

If it remains a constant value it will still cause issues for iOS-like platforms as described in: https://github.com/dotnet/runtime/issues/87924

My last comment: https://github.com/dotnet/runtime/issues/87924#issuecomment-1628821955 breaks down the benefits of having these conditional variables as feature switches.

Reply:
I think that would be a good enhancement to make in a future PR. It isn't necessary to fix the issue this PR is fixing - https://github.com/dotnet/runtime/issues/81803.